File: openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/SYSTEM/procedures/interactive.py
from openc3.script import *

INFORMATIVE = "This is informative"
DETAILS = "Additional details"

# This script checks all the interactive APIs
prompt("Would you like to continue?")
answer = combo_box("This is a plain combo box", "one", "two", "three", informative=None)
print(f"answer:{answer}")
answer = combo_box(
    "This is a combo box with info",
    "one",
    "two",
    "three",
    informative=INFORMATIVE,
)
print(f"answer:{answer}")
answer = combo_box(
    "This is a combo box with details",
    "one",
    "two",
    "three",
    informative=None,
    details=DETAILS,
)
print(f"answer:{answer}")
answer = combo_box(
    "This is a combo box with info & details",
    "one",
    "two",
    "three",
    informative=INFORMATIVE,
    details=DETAILS,
)
print(f"answer:{answer}")
answer = combo_box(
    "This is a combo box",
    "one",
    "two",
    "three",
    details=DETAILS,
)
print(f"answer:{answer}")
answer = combo_box("This is a multi-select combo box", "one", "two", "three", multiple=True)
print(f"answer:{answer} len:{len(answer)}")
answer = check_box("This is a check box", "one", "two", "three")
print(f"answer:{answer} len:{len(answer)}")
answer = check_box(
    "This is a check box with info & details",
    "one",
    "two",
    "three",
    informative=INFORMATIVE,
    details=DETAILS,
)
print(f"answer:{answer} len:{len(answer)}")
answer = prompt(
    "This is a test",
    details=DETAILS,
)
print(f"answer:{answer}")
answer = prompt("This is a test", details=DETAILS)
print(f"answer:{answer}")
answer = message_box(
    "This is a message box",
    "one",
    "two",
    "three",
    informative="Informative stuff",
    details=DETAILS,
)
print(f"answer:{answer}")
answer = vertical_message_box(
    "This is a message box",
    "one",
    "two",
    "three",
    informative="Informative stuff",
    details=DETAILS,
)
print(f"answer:{answer}")
QUESTION = "Let me ask you a question"
answer = ask(QUESTION, "default")
print(f"answer:{answer} type:{type(answer)}")
if not isinstance(answer, str):
    raise RuntimeError("Not a string")
answer = ask(QUESTION, 10)
print(f"answer:{answer} type:{type(answer)}")
if not isinstance(answer, int):
    raise RuntimeError("Not an integer")
answer = ask(QUESTION, 10.5)
print(f"answer:{answer} type:{type(answer)}")
if not isinstance(answer, float):
    raise RuntimeError("Not a float")
answer = ask_string(QUESTION, "default")
print(f"answer:{answer} type:{type(answer)}")
answer = ask_string(QUESTION, 10)
print(f"answer:{answer} type:{type(answer)}")
if not isinstance(answer, str):
    raise RuntimeError("Not a string")
answer = ask("Enter a blank (return)", True)  # allow blank
print(f"answer:{answer}")
answer = ask("Password", False, True)  # password required
print(f"answer:{answer}")

File: openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/SYSTEM/procedures/interactive.rb
# This script checks all the interactive APIs
prompt("Would you like to continue?")
answer = combo_box("This is a plain combo box", 'one', 'two', 'three', informative: nil)
puts "answer:#{answer}"
answer = combo_box("This is a combo box with info", 'one', 'two', 'three', informative: 'This is informative')
puts "answer:#{answer}"
answer = combo_box("This is a combo box with details", 'one', 'two', 'three', informative: nil, details: 'This is some details')
puts "answer:#{answer}"
answer = combo_box("This is a combo box with info & details", 'one', 'two', 'three', informative: 'This is informative', details: 'Details details details!')
puts "answer:#{answer}"
answer = combo_box("This is a combo box", 'one', 'two', 'three', details: "Some more stuff")
puts "answer:#{answer}"
answer = combo_box("This is a multi-select combo box", 'one', 'two', 'three', multiple: true)
puts "answer:#{answer} len:#{answer.length} [0]:#{answer[0]}"
answer = check_box("This is a check box", 'one', 'two', 'three')
puts "answer:#{answer} len:#{answer.length} [0]:#{answer[0]}"
answer = check_box("This is a check box with info & details", 'one', 'two', 'three', informative: 'This is informative', details: 'Details details details!')
puts "answer:#{answer} len:#{answer.length} [0]:#{answer[0]}"
answer = prompt("This is a test", details: "Some more stuff")
puts "answer:#{answer}"
answer = prompt("This is a test", details: "Some more stuff")
puts "answer:#{answer}"
answer = message_box('This is a message box', 'one', 'two', 'three', informative: "Informative stuff", details: "Some more stuff")
puts "answer:#{answer}"
answer = vertical_message_box('This is a message box', 'one', 'two', 'three', informative: "Informative stuff", details: "Some more stuff")
puts "answer:#{answer}"
answer = ask("Let me ask you a question", "default")
puts "answer:#{answer} class:#{answer.class}"
raise "Not a string" unless answer.is_a? String
answer = ask("Let me ask you a question", 10)
puts "answer:#{answer} class:#{answer.class}"
raise "Not an integer" unless answer.is_a? Integer
answer = ask("Let me ask you a question", 10.5)
puts "answer:#{answer} class:#{answer.class}"
raise "Not a float" unless answer.is_a? Float
answer = ask_string("Let me ask you a question", "default")
puts "answer:#{answer} class:#{answer.class}"
answer = ask_string("Let me ask you a question", 10)
puts "answer:#{answer} class:#{answer.class}"
raise "Not a string" unless answer.is_a? String
answer = ask("Enter a blank (return)", true) # allow blank
puts "answer:#{answer}"
answer = ask("Password", false, true) # password required
puts "answer:#{answer}"
